spec/ddtrace/contrib/rails/middleware_spec.rb

```diff
@@ -76,6 +76,40 @@ def call(env)
       end
     end
 
+    context 'that itself creates a span' do
+      let(:middleware) do
+        stub_const('CustomSpanMiddleware', Class.new do
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            Datadog.tracer.trace('custom.test') do
+              @app.call(env)
+            end
+          end
+        end)
+      end
+
+      context 'and added after tracing is enabled' do
+        before(:each) do
+          custom_span_middleware = middleware
+          rails_test_application.configure { config.app_middleware.use custom_span_middleware }
+        end
+
+        context 'with #middleware_names' do
+          let(:use_rack) { false }
+          let(:rails_options) { super().merge!(middleware_names: true) }
+
+          it do
+            get '/'
+            span = all_spans.find { |s| s.name == 'rack.request' }
+            expect(span.resource).to eq('TestController#index')
```

Comment:
Should we also assert other parts of the expected trace here too? e.g. number of spans, name of spans, parent hierarchy or spans? Just to verify the trace is a good representation of what we expect.

Something like:

```ruby
let(:rack_span) { all_spans[2] }
let(:middleware_span) { all_spans[1] }
let(:controller_span) { all_spans[0] }

it do
  get '/'

  expect(all_spans).to have(3).items
  
  expect(rack_span.name).to eq('rack.request')
  expect(rack_span.resource).to eq('TestController#index')
  expect(rack_span.parent_id).to eq(0)

  expect(middleware_span.name).to eq('custom.test')
  expect(middleware_span.parent_id).to eq(rack_span.span_id)

  expect(controller_span.name).to eq('rails.action_controller')
  expect(controller_span.parent_id).to eq(middleware_span.span_id)
end
```

Reply:
I'd rather not specify stuff that's not related to the aspect of the code being tests, which in this case is that the `rack.request` span gets the correct resource set. But I can add that if needed?